src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -122,9 +122,10 @@ protected void handlePut(RestChannel channel, final RestRequest request, final C
 
         // if password is set, it takes precedence over hash
         String plainTextPassword = additionalSettingsBuilder.get("password");
-        if (plainTextPassword != null && plainTextPassword.length() > 0) {
+        if (plainTextPassword != null) {
             additionalSettingsBuilder.remove("password");
-            additionalSettingsBuilder.put("hash", hash(plainTextPassword.toCharArray()));
+            if (plainTextPassword.length() > 0)
+                additionalSettingsBuilder.put("hash", hash(plainTextPassword.toCharArray()));
```

Comment:
If empty password is not allowed then we should fail the request instead of ignoring it. Empty password if set may be considered as valid password and hash can be generated for empty string.